### ISSUE TITLE
zeta(1/2) returns nan

### ISSUE THREAD

**Issue by simaki** [pytorch/pytorch#68110]
## Bug

The [Riemann zeta function](https://pytorch.org/docs/stable/special.html?highlight=zeta#torch.special.zeta) at 1/2 should be `zeta(1/2) = -1.46...` (See [WolframAlpha](https://www.wolframalpha.com/input/?i=zeta%281%2F2%29)) but [`torch.special.zeta`](https://pytorch.org/docs/stable/special.html?highlight=zeta#torch.special.zeta) returns `nan`.

This may be expected behavior but if so it would be nice to have documentation for this behavior.

## To Reproduce

```py
import torch

output = torch.special.zeta(torch.tensor(1 / 2), 1.0)

print(output)
```

```txt
tensor(nan)
```

## Expected behavior

```txt
tensor(-1.46...)
```

## Environment

```txt
Collecting environment information...
PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 11.6 (x86_64)
GCC version: Could not collect
Clang version: 12.0.5 (clang-1205.0.22.11)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.7 (default, Oct 13 2021, 06:45:31)  [Clang 13.0.0 (clang-1300.0.29.3)] (64-bit runtime)
Python platform: macOS-11.6-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy==0.910
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.2
[pip3] torch==1.10.0
[conda] Could not collect
```

**Comment by mruberry** [comment]
Thanks for reporting this issue, @simaki

**Comment by kshitij12345** [comment]
On digging a bit,

```python
>>> import sciy.special
>>> scipy.special.zeta(1/2)
-1.4603545088095866
>>> scipy.special.zeta(1/2, 1.)
nan
```

This happens due to dispatch to different function depending on whether `q` is specified.
https://github.com/scipy/scipy/blob/e32fc2329d3dd23298725153c5b2cc7fcd0f14f1/scipy/special/_basic.py#L2577-L2580

Will look a bit more into that.

**Comment by Lezcano** [comment]
Note that the value that you provide at that point is not the value of the sum (which is what the documentation mentions it is computed) but the value of the **analytic continuation** of the function when seen as a meromorphic function via its functional equation.

At the moment, we do not even support complex numbers on `special`, so I do not think it makes much sense to talk about returning the analytic continuation of the function, as the values of the continuation of the zeta function onto the whole real axis could very well not be real. In the case of the zeta function, you can in fact define \zeta : R -> R, but, for example, the gamma function you have that \gamma : R -> C, as it has poles on the negative numbers.

What I would is:
- First, document the domain of the functions in `special`.
- Then, address https://github.com/pytorch/pytorch/issues/67333 and extend the support of functions in `special` to take complex inputs.
- Finally, implement (and document) analytic extensions of functions in special when reasonable. For example, for the zeta function, one could do this recursively on stripes of width 1, in terms of the values for Re(z) > 1, although there surely are better formulas.

**Comment by essandess** [comment]
> Note that the value that you provide at that point is not the value of the sum (which is what the documentation mentions it is computed) but the value of the **analytic continuation** of the function when seen as a meromorphic function via its functional equation. … As a side note, I believe we should return `inf` rather than `NaN` whenever Re(z) <= 1.

This would be incorrect both mathematically and programmatically.

Mathematically, ζ(*s*) is well defined for all real *s* ≠ 1, whether via analytic continuation, reflection, or many other properties of the zeta function. This fact is reflected in `scipy`'s underlying `xsf` code:

https://github.com/scipy/xsf/blob/963914a33efa4a3a4ff27e236827ea66e5107dd8/include/xsf/cephes/zetac.h#L263-L275

Indeed,
```python
import scipy.special as sps

sps.zeta(1/2)
-1.4603545088095866
```

`torch`'s zeta Riemann function and documentation should all correspond to this fact over the entire range of the Riemann zeta function. Programatically, the extension cases are somewhat trivial.
